src/enclave/connection.ts

```diff
@@ -91,10 +92,14 @@ export class Enclave implements EnclaveConnection {
 		return this.sendCall(tx) as Promise<TxReceipt>;
 	}
 
+	public async trade(tx: Trade): Promise<TxReceipt> {
+		return this.sendCall(tx) as Promise<TxReceipt>;
+	}
+
 	public async exit(exitRequest: ExitRequest): Promise<BalanceProof> {
 		const p = new Promise<BalanceProof>((resolve, reject) => {
 			this.once("exitproof", resolve);
-			this.sendCall(exitRequest).catch(reject);
+			this.sendCall(exitRequest).catch(err => reject(new Error(err)));
```

Comment:
`err` is not guaranteed to be a  `string`. Shouldn't this die when an object is fed to the `Error` constructor, since it is defined as:

```typescript
interface ErrorConstructor {
    new(message?: string): Error;
    (message?: string): Error;
    readonly prototype: Error;
}
```

I think a `instanceof` or similar is needed here. I do not know how this behaves on user-side.

Reply:
Same for every other occurrence of `new Error(...)`.

Reply:
so maybe keep as it was before? why the change @RmbRT ?

Reply:
during testing, I had errors, where promises were rejected with non-`Error` objects, and typescript requested me kindly to please only throw `Error`s.

Reply:
ok then do an `instanceof Error` check like `return e instanceof Error ? e : new Error(e)` and hope that it's a string ^^